src/Analyser/TypeSpecifier.php

```diff
@@ -326,6 +326,50 @@ public function specifyTypesInCondition(
 						return $this->create($exprNode->getArgs()[0]->value, $type, $context, false, $scope, $rootExpr);
 					}
 				}
+
+				if (
+					!$context->null()
+					&& $exprNode instanceof FuncCall
+					&& $exprNode->name instanceof Name
+					&& strtolower((string) $exprNode->name) === 'get_parent_class'
+					&& isset($exprNode->getArgs()[0])
+					&& $constantType instanceof ConstantStringType
+					// the is_string() check should not be necessary, since ConstantStringType->getValue() returns 'string'
+					&& is_string($constantType->getValue())
```

Comment:
this works arround the type error:

![grafik](https://user-images.githubusercontent.com/120441/192321661-a3f2db1b-af7d-4453-ac30-f119002b7fb7.png)

which is a separate bug, see https://github.com/phpstan/phpstan-src/pull/1396

Reply:
in https://github.com/phpstan/phpstan-src/pull/1383/commits/e91e1f1e62b9d4533eaf65852f91e1f878e94392 I factored out the existing logic into 2 new private methods. 

that way we don't run into the https://github.com/phpstan/phpstan-src/pull/1396 bug and don't need the `is_string` based workaround

Reply:
fyi @rvanvelzen, the refactoring might conflict with your existing https://github.com/phpstan/phpstan-src/pull/1761 PR.